src/lib.rs

```diff
@@ -672,7 +672,18 @@ pub unsafe extern "C" fn fd_readdir(
 /// would disappear if `dup2()` were to be removed entirely.
 #[no_mangle]
 pub unsafe extern "C" fn fd_renumber(fd: Fd, to: Fd) -> Errno {
-    unreachable()
+    State::with_mut(|state| {
+        let closed = state.closed;
+
+        let fd_desc = state.get_mut(fd)?;
+        let desc = replace(fd_desc, Descriptor::Closed(closed));
+
+        let to_desc = state.get_mut(to)?;
+        *to_desc = desc;
```

Comment:
I think this'll want to do a `replace` of `to_desc` and the close the result if it's a file/socket?

Reply:
Arg, instead of me always forgetting to drop these, I've now implemented `Drop` for `Descriptor` to do this.